### PR TITLE
feat: card carousel

### LINKED ISSUE #203: 404 page needs to be responsive
### Problem
The 404 page is not responsive creating issues on smaller devices.

### Solution
The design should be `display: block` or `flex-direction: column` on smaller resolutions.

### Proposed Solution
Since Tailwind is a mobile first framework, the flex container might be best used only after a certain size. Example: `md:flex`

This would be a great first issue for someone to get involved!

### LINKED ISSUE #340: ⚡️ 🏠 📄 ⚡️ Card Carousel component
### Card Carousel component

We want to replace the current card section with a scrolling carousel from the [Figma Design File](https://www.figma.com/file/nLwxUWpunk8SgvjVpyE2gd/MHF-Boards?node-id=0%3A1). Whoever gets assigned should be in communication as developing this to discuss how to integrate certain sections. For instance the buttons should only appear when either hovering on the card, or when the card enters the center of the carousel. 

Screen Shot 2021-11-09 at 1 54 01 PM

### LINKED ISSUE #343: Extra horizontal space on our website
**What happened?**
* Overflow of content horizontally and extra space present.
* Bug observed in browsers: Google Chrome, Firefox, Brave. Operating System Ubuntu 20.10.
 Screen Resolution > 2088*600
Screenshot from 2021-11-16 00-22-59

**Steps to reproduce**
1. Open [Help a family main website](https://helpafamily.margaritahumanitarian.org/).
2. Scroll horizontally and see the extra space.

**What did you expect to happen?**
* No horizontal scroll should be there.

### PR DESCRIPTION
**Pull request by akshitarora921**
### This PR: 
Add carousel from cards
fixes #203 
fixes #343 
### Self-Review Checklist
- [x] PR title is clear with proper spelling and grammar
- [x] PR description contains a bulleted list of changes contained in the PR
- [x] PR links to relevant issues, with keywords to auto-close any issues fully resolved upon merge
- [x] All automated checks passed
- [x] Any variables introduced are named clearly and explicitly

### CODE REVIEW

**Comment by akshitarora921** [comment]
Pending
- [ ] Mobile layout
- [ ] Prop type definition (If required)

File: pages/index.js [hunk 1]
```diff
@@ -26,16 +27,19 @@ export const getStaticProps = async () => {
 export default function Home({ image }) {
   return (
     <PrimaryLayout image={image} inNeed main>
-      <HotMealDaySection />
+      {/* <HotMealDaySection /> */}
```
Comment (LowLifeArcade) [inline pages/index.js:30]: Why did you remove this section?

**Review by LowLifeArcade — CHANGES_REQUESTED** [review]
Hey @akshitarora921. Great job with all of your hard work. It's very much appreciated. There's a lot to go through but I'll start very simply. First question is about the hotmealday section. Not sure what you were going for by removing it. It should still be there.

File: components/Card.js [hunk 2]
```diff
@@ -21,56 +42,91 @@ function Card({ backgroundImageSource, backgroundImageAltText, children }) {
         </figure>
       )}
       <div
-        className={`card-body grid ${cardsBackgroundColor} text-${textColor} gap-y-3 auto-rows-card`}
+        className={`carousel-body flex flex-col p-6 shadow-lg justify-start items-center ${cardsBackgroundColor} text-${textColor} rounded-b-md`}
```
Comment (LowLifeArcade) [inline components/Card.js:45]: I'm curious why you changed the class card-body to carousel-body

File: components/Card.js [hunk 3]
```diff
@@ -21,56 +42,91 @@ function Card({ backgroundImageSource, backgroundImageAltText, children }) {
...
         .card-shadow {
           box-shadow: rgba(14, 30, 37, 0.061) 6px 6px 12px 0px,
             rgba(14, 30, 37, 0.075) 6px 6px 10px 0px;
         }
+        .carousel-card {
+          min-width: 350px;
+          max-width: 350px;
+        }
+        .carousel-body {
+          min-height: 420px;
```
Comment (LowLifeArcade) [inline components/Card.js:84]: I think this is a good number as per the design file, but the problem is that it's not including the image size. The overall card (carousel-body) should be this size, but this is the image size + card (carousel-body). I recommend changing the class name to card-body and making it account for the whole card body including the image.

**Review by LowLifeArcade — CHANGES_REQUESTED** [review]
I think you're doing a good job @akshitarora921. There are a few things we can work on and we'll do it piece by piece. 

But first of all, I want to point out that the figma file has dimensions for the components. It looks like you may have referenced them, but if you haven't I recommend using them as they are the quickest way to success here.

Screen Shot 2021-11-13 at 3 34 21 PM

File: components/CardsLayout.js [hunk 4]
```diff
@@ -1,13 +1,49 @@
...
-      <div className="w-lg">
-        <h2 className="pb-20 text-3xl md:text-4xl">{description}</h2>
-        <div className="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-3 md:gap-x-8 gap-y-5 justify-center align-center content-center">
-          {children}
-        </div>
+    <div className="w-full py-20 relative ">
+      <h2 className="pb-20 text-3xl md:text-4xl z-10 text-center">
+        {description}
+      </h2>
+      <div className="absolute top-0 left-0 h-full w-full overflow-hidden flex justify-center items-center z-0">
```
Comment (LowLifeArcade) [inline components/CardsLayout.js:15]: This looks like it works almost perfectly. Maybe we can make the size scale with an `object-fit: cover`, or some similar html attribute that will make it scale with larger screen sizes but has a `min-width` so it holds its size otherwise.

File: components/CardsLayout.js [hunk 5]
```diff
@@ -1,13 +1,49 @@
...
+        aria-label="move left"
+        className={`btn btn-accent rounded-sm btn-size absolute left-4 top-1/2 z-50 ${
+          !hasItemsOnLeft && 'hidden'
+        }`}
+        // todo: -translate-y-1/2 this also we need to add to make it vertically center
+        // disabled={isPending ? true : false}
+        onClick={scrollLeft}
+        type="button"
+      >
+        <AiOutlineLeft />
```
Comment (LowLifeArcade) [inline components/CardsLayout.js:40]: I like the button use here. The way they appear and operate is very nice.

File: components/ElectronicsDropOffCard.js [hunk 6]
```diff
@@ -1,27 +1,30 @@
...
-      <CardAddress label="Email us to learn more at:">
-        {
-          <a href="mailto:hi@mhfcares.org?subject=Electronics%20Drop-Off%20Box">
-            {'hi@mhfcares.org'}
-          </a>
-        }
-      </CardAddress>
-    </Card>
+      paragraph={
+        <CardParagraph>
```
Comment (LowLifeArcade) [inline components/ElectronicsDropOffCard.js:20]: This component seems to only wrap its children in a span tag. Maybe we can refactor here and anywhere else this is being used.

File: hooks/usePosition.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
+import { useCallback, useEffect, useState } from 'react';
+
+export function usePosition(ref) {
```
Comment (LowLifeArcade) [inline hooks/usePosition.js:3]: You spent a lot of time here. Nice work!

File: utils/isAvailable.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+export function isAvailable(value) {
```
Comment (LowLifeArcade) [inline utils/isAvailable.js:1]: Maybe we can rename this to be more clear.

File: pages/index.js (hunk 1 above)
Comment (akshitarora921) [reply pages/index.js:30]: They were separate components and have similar styles so, I have created a `cardsData.json` and map over it.

File: utils/isAvailable.js (hunk 8 above)
Comment (akshitarora921) [reply utils/isAvailable.js:1]: Sure, Can you suggest a better name?

File: components/ElectronicsDropOffCard.js (hunk 6 above)
Comment (akshitarora921) [reply components/ElectronicsDropOffCard.js:20]: Yeah, so these components whose names are ending with the card, I was trying to make them generic and then I put all their data in JSON file. So, my final plan is to delete them as we have the data.

File: components/CardsLayout.js (hunk 5 above)
Comment (akshitarora921) [reply components/CardsLayout.js:40]: Thanks :)

File: components/CardsLayout.js (hunk 4 above)
Comment (akshitarora921) [reply components/CardsLayout.js:15]: I will try.

File: components/Card.js (hunk 3 above)
Comment (akshitarora921) [reply components/Card.js:84]: sure!

File: components/Card.js (hunk 2 above)
Comment (akshitarora921) [reply components/Card.js]: now, I also don't remember 😆

File: pages/index.js (hunk 1 above)
Comment (LowLifeArcade) [reply pages/index.js:30]: Let's put this section back as it's part of the design.

Comment (LowLifeArcade) [reply pages/index.js:30]:
```suggestion
      <HotMealDaySection />
```

File: pages/index.js [hunk 9]
```diff
@@ -1,18 +1,19 @@
...
-import HotMealDaySection from '../components/HotMealDaySection';
-import HumanitarianClinicCard from '../components/HumanitarianClinicCard';
-import HygieneKitsCard from '../components/HygieneKitsCard';
-import LaptopForFamiliesCard from '../components/LaptopForFamiliesCard';
+// import ElectronicsDropOffCard from '../components/ElectronicsDropOffCard';
+// import FallPreventionForEldersCard from '../components/FallPreventionForEldersCard';
+// import FoodDistributionSupportCard from '../components/FoodDistributionSupportCard';
+// import HealthWorkshopCard from '../components/HealthWorkshopCard';
+// import HotMealDayCard from '../components/HotMealDayCard';
+// import HotMealDaySection from '../components/HotMealDaySection';
```
Comment (LowLifeArcade) [inline pages/index.js:10]:
```suggestion
import HotMealDaySection from '../components/HotMealDaySection';
```

File: components/Card.js (hunk 3 above)
Comment (LowLifeArcade) [reply components/Card.js]: Just to keep a note of our conversation. We'll change the content that seems to be creating overflow issues here after the PR is pushed through. Let's just get to a place where we don't have the cards jumping in size 🦖

File: components/Card.js [hunk 10]
```diff
@@ -1,13 +1,28 @@
 import Image from 'next/image';
-import PropTypes from 'prop-types';
 import React from 'react';
 import clsx from 'clsx';
+import { isAvailable } from 'utils/isAvailable';
 import { useContextTheme } from 'components/ThemeContext';
+import useStripeSession from 'hooks/useStripeSession';
```
Comment (LowLifeArcade) [inline components/Card.js:7]:
```suggestion
// test
```

**Review by LowLifeArcade — CHANGES_REQUESTED** [review]

**Comment by LowLifeArcade** [comment]
Smol checklist 
- [ ] put Our Humanitarian Programs in indexjs instead of component 
- [ ] and fix background svg accordingly
- [ ] create separate json file per page

**Comment by akshitarora921** [comment]
Title Our Humanitarian Programs is still part of CardLayout as now it will not affect the background

**Comment by LowLifeArcade** [comment]
Looks great @akshitarora921. There's a few things we can address in another issue but this was well done. Merging! 🦖 ✨

File: pages/index.js [hunk 11]
```diff
@@ -26,16 +18,10 @@ export const getStaticProps = async () => {
 export default function Home({ image }) {
   return (
     <PrimaryLayout image={image} inNeed main>
-      <HotMealDaySection />
```
Comment (LowLifeArcade) [inline pages/index.js:29]: Why did you take this out again??

**Comment by akshitarora921** [comment]
😭😁

File: pages/index.js (hunk 11 above)
Comment (akshitarora921) [reply pages/index.js:29]: Sorry, I miss read it by <HotMealDayCard/>
